controllers/storagecluster/cephtoolbox.go

```diff
@@ -117,21 +119,36 @@ func removeDuplicateTolerations(tolerations []corev1.Toleration) []corev1.Tolera
 	return list
 }
 
-func getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
+func (r *StorageClusterReconciler) getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
 	err := validateMultusSelectors(sc.Spec.Network.Selectors)
 	if err != nil {
 		return "", err
 	}
 
-	multusNetNs, err := util.GetOperatorNamespace()
-	if err != nil {
-		return "", err
+	if sc.Spec.Network.Selectors["public"] == "" {
+		return "", nil
 	}
 
-	if sc.Spec.Network.Selectors["public"] != "" {
-		multusNetName := sc.Spec.Network.Selectors["public"]
-		return fmt.Sprintf("%s/%s", multusNetNs, multusNetName), nil
+	multusNetName := sc.Spec.Network.Selectors["public"]
+	multusNetNamespacedName := strings.Split(multusNetName, "/")
+	nadNS, nadName := "", ""
+	if len(multusNetNamespacedName) == 2 {
```

Comment:
Let's change the order of the if conditions as well. First ==1 and then ==2 and then else.

Reply:
Done